### DISCUSSION TITLE
The result of router.isFallback changes true to false

### DISCUSSION THREAD

**Discussion by HeppokoNeet** [vercel/next.js#36217] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

|Environment|Version|
|:--|:--:|
|Rails|7.0.0|
|React|18.0|
|Next.js|12.1.4|

### What browser are you using? (if relevant)

Chrome 100.0.4896.75

### How are you deploying your application? (if relevant)

next build

### Describe the Bug

My setting of `getStaticPaths` and `getStaticProps` are here:
```
export const getStaticPaths = async () => {
  const data = await axios.get(
    `${process.env.NEXT_PUBLIC_ENDPOINT}/admin/users_data`
  );

  const paths = data.data.users.map((user: any) => {
    return { params: { id: user.id.toString() } };
  });
  return {
    paths,
    fallback: true,
  };
};

export const getStaticProps = async (params: any) => {
  const users = params.params.id;
  return {
    props: { users },
    revalidate: 10,
  };
};
```

I want to add users and can access the detail page after `next build`.
<br>
I heard that I can achieve it using `revalidate` and `fallback: true`.

I wrote an error option if there is no path.
```
import ErrorPage from "next/error";

export default function UsersDetailPage(props: {
  users: any;
}) {
  const router = useRouter();

  return !router.isFallback && props.users ? (
// UsersDetailPage
  ) : (
    <ErrorPage statusCode={404} />
  );
}
```

From this code when I access to wrong path directly, showing an error page in few seconds.
<br>
parameter is this:
```
props.users: undefined
router.isFallback: true
```
But after that, parameter will change to
```
props.users: example // <-- not existing
router.isFallback: false
```

And show the detail page.
<br>
I can't understand what's going on.

Can you teach me how I can solve the problem?
<br>
Thank you.

## Gif

![名称未設定-1](https://user-images.githubusercontent.com/36767393/163674376-69c261ac-126a-4e8d-8d9c-be312197a8b3.gif)

### Expected Behavior

Not changing router.isFallback `true` to  `false` and keep showing 404 page.

### To Reproduce

I've already tried:

`fallback: "blocking"`
`{ notFound: true }` in JSX

**Comment by ijjk** [comment]
Hi, `router.isFallback` is `true` when showing the fallback page (loading page) while the page is generated in the background, this is set to `false` once the data is loaded and props have been updated.

**Comment by icyJoseph** [comment]
Let's simplify a bit:

- When a visible `next/link` points to a page using `fallback: true` with `getStaticPaths`, that is, ISR, the page is pre-rendered by the runtime Next.js server, so that when the user clicks on it the page already exists.
- However, if you navigate to a non previously generated page, and fallback is true, not "blocking", then Next.js responds and gives you the ability to show a spinner, check [here](https://github.com/icyJoseph/next-js-react-gbg/blob/main/demo-app/src/pages/pokemon/%5Bid%5D.tsx).

It is possible of course for the first case, to not be done yet, and upon navigation a spinner is still shown, but less likely.

Bots, will not see the loading state, for SEO reasons, and because bots don't care about loading states I guess.

**Comment by nghiepdev** [comment]
Tested `12.1.4`. When using `fallback: true`, the `router.isFallback` always is `true` in the development and changes to `false`.
